### ISSUE TITLE
Python 3.9 debug cp39d bad argument to internal function

### ISSUE THREAD

**Issue by inevity** [ijl/orjson#144]
import orjson
orjson.dumps([])  // here met issue

call src/serialize/writer.rs resize()

```
        unsafe {
            _PyBytes_Resize(
                &mut self.bytes as *mut *mut PyBytesObject as *mut *mut PyObject,
                len as isize,
            );
        }
```
The _PyBytes_Resize return above error. 
Python 3.9 debug with py_debug and with_trace_refs ,so it is 39d. 
Versions: orjson master, maturin master, pyo3 0.12.3. 
Use pip wheel build ,maturin build ,same issue.

**Comment by ijl** [comment]
I haven't used a debug build. Is this also happening with a non-debug build? If so `python -m sysconfig` would be helpful.

**Comment by inevity** [comment]
Have not test the non-debug build.

**Comment by inevity** [comment]
Found that the json/enum/uuid type init func lead this issue. It is absent of the incref for the funcobject.globals, but once gc is enabled or  when app finished with gc disabled, the cpython func_clear will decref the globals. Then once gc collect run , will find this error. 
Pr will be availiable.

**Comment by inevity** [comment]
Issue "gc_decref: Assertion "gc_get_refs(g) > 0" failed: refcount is too small" have been fixed,
But bad argument to internal function is another issue remaining unsolved

**Comment by ijl** [comment]
@inevity if you have time to open a pull request I'll look at incorporating it.

**Comment by inevity** [comment]
> @inevity if you have time to open a pull request I'll look at incorporating it.

Please review.

**Comment by ijl** [comment]
Fixed in 3.4.6: https://github.com/ijl/orjson/releases/tag/3.4.6
